databind/src/main/java/gov/nist/secauto/metaschema/databind/io/json/MetaschemaJsonReader.java

```diff
@@ -711,7 +722,12 @@ public List<?> readList() throws IOException {
         IBoundInstanceFlag jsonKey = instance.getItemJsonKey(item);
         assert jsonKey != null;
 
-        String key = ObjectUtils.requireNonNull(jsonKey.getValue(item)).toString();
+        Object keyValue = jsonKey.getValue(item);
+        if (keyValue == null) {
+          throw new IOException(String.format("Null JSON key value for definition '%s'",
```

Comment:
Minor, non-blocking, we are just talking about the value of `json-key` here, right? I thought using its Metaschema syntax name would make it clear (to me, at least).

```suggestion
          throw new IOException(String.format("Null value for json-key for definition '%s'",
```